contracts/extensions/contracts/AllocationExtension.sol

```diff
@@ -0,0 +1,141 @@
+
+abstract contract AllocationExtension is CoreBaseStrategy, IAllocationExtension {
+    /// ================================
+    /// ========== Storage =============
+    /// ================================
+
+    /// @notice The start and end times for allocations
+    uint64 public allocationStartTime;
+    uint64 public allocationEndTime;
+
+    /// @notice Defines if the strategy is sending Metadata struct in the data parameter
+    bool public isUsingAllocationMetadata;
+
+    /// @notice token -> isAllowed
+    mapping(address => bool) public allowedTokens;
+
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    /// @notice This initializes the Alocation Extension
+    /// @dev This function MUST be called by the 'initialize' function in the strategy.
+    /// @param _allowedTokens The allowed tokens
+    /// @param _allocationStartTime The start time for the allocation period
+    /// @param _allocationEndTime The end time for the allocation period
+    /// @param _isUsingAllocationMetadata Defines if the strategy is sending Metadata struct in the data parameter
+    function __AllocationExtension_init(
+        address[] memory _allowedTokens,
+        uint64 _allocationStartTime,
+        uint64 _allocationEndTime,
+        bool _isUsingAllocationMetadata
+    ) internal virtual {
+        if (_allowedTokens.length == 0) {
+            // all tokens
+            allowedTokens[address(0)] = true;
+        } else {
+            for (uint256 i; i < _allowedTokens.length; i++) {
+                allowedTokens[_allowedTokens[i]] = true;
+            }
+        }
+
+        isUsingAllocationMetadata = _isUsingAllocationMetadata;
+
+        _updateAllocationTimestamps(_allocationStartTime, _allocationEndTime);
+    }
+
+    /// ====================================
+    /// =========== Modifiers ==============
+    /// ====================================
+
+    /// @notice Modifier to check if allocation has ended
+    /// @dev Reverts if allocation has not ended
+    modifier onlyAfterAllocation() {
+        _checkOnlyAfterAllocation();
+        _;
+    }
+
+    /// @notice Modifier to check if allocation is active
+    /// @dev Reverts if allocation is not active
+    modifier onlyActiveAllocation() {
+        _checkOnlyActiveAllocation();
+        _;
+    }
+
+    /// @notice Modifier to check if allocation has started
+    /// @dev Reverts if allocation has started
+    modifier onlyBeforeAllocation() {
+        _checkBeforeAllocation();
+        _;
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Checks if the allocator is valid
+    /// @param _allocator The allocator address
+    /// @return 'true' if the allocator is valid, otherwise 'false'
+    function _isValidAllocator(address _allocator) internal view virtual returns (bool);
```

Comment:
I think we can remove this, it should be part of the allowlist extension

Reply:
My idea is that maybe you need `_isValidAllocator` but dont need all other logic from allowlist extension. 
For example in: `DirectGrantsSimpleStrategy`, `MicroGrantsGovStrategy`, `MicroGrantsHatsStrategy`, and a few other strats

Reply:
oke sounds good then

Reply:
What about implementing it and returning `true` as default so that strategies not using `_isValidAllocator()` don't have to implement it.